homeassistant/components/tessie/__init__.py

```diff
@@ -94,41 +95,48 @@ async def async_setup_entry(hass: HomeAssistant, entry: TessieConfigEntry) -> bo
 
     # Energy Sites
     tessie = Tessie(session, api_key)
+    energysites: list[TessieEnergyData] = []
+
     try:
-        products = (await tessie.products())["response"]
+        scopes = await tessie.scopes()
     except TeslaFleetError as e:
         raise ConfigEntryNotReady from e
 
-    energysites: list[TessieEnergyData] = []
-    for product in products:
-        if "energy_site_id" in product:
-            site_id = product["energy_site_id"]
-            api = EnergySpecific(tessie.energy, site_id)
-            energysites.append(
-                TessieEnergyData(
-                    api=api,
-                    id=site_id,
-                    live_coordinator=TessieEnergySiteLiveCoordinator(hass, api),
-                    info_coordinator=TessieEnergySiteInfoCoordinator(hass, api),
-                    device=DeviceInfo(
-                        identifiers={(DOMAIN, str(site_id))},
-                        manufacturer="Tesla",
-                        name=product.get("site_name", "Energy Site"),
-                    ),
+    if Scope.ENERGY_DEVICE_DATA in scopes:
+        try:
+            products = (await tessie.products())["response"]
+        except TeslaFleetError as e:
+            raise ConfigEntryNotReady from e
+
+        for product in products:
+            if "energy_site_id" in product:
```

Comment:
This IMHO is not a good pattern, the problem with this, is that users have no idea why information is missing.

Reply:
It's not a repair issue if a user has chosen to not provide that scope to Tessie, many people do this by choice.

I do guess they could then choose to ignore it, but I assumed I should respect the choice they already made when signing up.

Reply:
Aah check, ok the scope if further upstream 👍 

makes sense, thanks for the context 👍